R/derive_interp_records.R

```diff
@@ -0,0 +1,74 @@
+#' Derive interpolated rows for the CDC charts (>=2 yrs old)
+#'
+#' Derive interpolated rows for the CDC charts (>=2 yrs old) by age in days
+#' for the following parameters: HEIGHT, WEIGHT and BMI
+#'
+#' @param dataset Input metadataset
+#'
+#'   The variables `AGE`, `AGEU`, `SEX`, `L`, `M`, `S` are expected to be in the dataset
+#'
+#'   For BMI the additional variables `P95` and `Sigma` are expected to be in the dataset
+#'
+#'   Note that `AGE` must be in days so that `AGEU` is equal to `"DAYS"`
+#'
+#' @param parameter CDC/WHO metadata parameter
+#'
+#' *Permitted Values*: `"WEIGHT"`, `"HEIGHT"` or `"BMI"` only - Must not be `NULL`
+#'
+#'   e.g. `parameter = "WEIGHT"`,
+#'        `parameter = "HEIGHT"`,
+#'   or   `parameter = "BMI"`.
+#'
+#' @return The input dataset plus additional interpolated records
+#'
+#' @family metadata
+#'
+#' @keywords metadata
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#'
+#' cdc_htage <- admiralpeds::cdc_htage %>%
+#'   mutate(
+#'     SEX = case_when(
+#'       SEX == 1 ~ "M",
+#'       SEX == 2 ~ "F",
+#'       TRUE ~ NA_character_
+#'     ),
+#'     AGE = round(AGE * 30.4375)
+#'   ) %>%
+#'   # Interpolate the AGE by SEX
+#'   group_by(SEX) %>%
```

Comment:
see my comment above - reminder to update example here also if you agree with the suggested update

Reply:
Done